tests/test_project/tests/test_selenium.py

```diff
@@ -0,0 +1,418 @@
+            account_button.click()
+            self.assertEqual(active_mg.is_displayed(), True)
+            mg_head = self._get_active_mg_head()
+            mg_head.click()
+            hamburger.click()
+            hamburger.click()
+            self.assertEqual(active_mg.is_displayed(), False)
+        self.web_driver.set_window_size(1366, 768)
+
+    def test_menu_on_medium_screen(self):
+        self.login()
+        self.web_driver.set_window_size(980, 600)
+        menu_toggle = self._get_menu_toggle()
+        menu_backdrop = self._get_menu_backdrop()
+        with self.subTest('Test menu remains close on first load'):
+            self._test_menu_state(False)
+            self.assertEqual(menu_backdrop.is_displayed(), False)
+        with self.subTest('Test menu gets open on clicking menu_toggle'):
+            menu_toggle.click()
+            self._test_menu_state(True)
+            self.assertEqual(menu_backdrop.is_displayed(), True)
+
+        with self.subTest('Test menu gets closed on clicking menu_backdrop'):
+            menu_backdrop.click()
+            self._test_menu_state(False)
+            self.assertEqual(menu_backdrop.is_displayed(), False)
+        self._open_menu()
+        self.web_driver.refresh()
+        WebDriverWait(self.web_driver, 2).until(
+            EC.visibility_of_element_located((By.XPATH, self.config['hamburger_xpath']))
+        )
+        with self.subTest('Test menu remains close on page change or refresh'):
+            self._test_menu_state(False)
+        self._test_account_component()
+        self._test_menu_dropdown(is_medium=True)
+        with self.subTest('Test menu on popup page'):
+            self._test_popup_page()
+        # Test active menu group
+        self._test_login_and_logout_page()
+        self.web_driver.set_window_size(1366, 768)
+
+    def test_menu_on_narrow_screen(self):
+        self.login()
+        self.web_driver.set_window_size(450, 600)
+        hamburger = self._get_hamburger()
+        menu_backdrop = self._get_menu_backdrop()
+        with self.subTest('Test menu remains close on first load'):
+            self._test_menu_state(False, is_narrow=True)
+            self.assertEqual(menu_backdrop.is_displayed(), False)
+        with self.subTest('Test opening of menu'):
+            hamburger.click()
+            self._test_menu_state(True, is_narrow=True)
+            self.assertEqual(menu_backdrop.is_displayed(), False)
+        self.web_driver.refresh()
+        with self.subTest('Test menu do not remain open on page change or refresh'):
+            self._test_menu_state(False, is_narrow=True)
+        self._test_account_component(is_narrow=True)
+        self._test_menu_dropdown(is_narrow=True)
+        with self.subTest('Test menu on popup page'):
+            self._test_popup_page()
+        self._test_login_and_logout_page()
+        self.web_driver.set_window_size(1366, 768)
+
+    def _test_login_and_logout_page(self, is_logged_in=True):
```

Comment:
Can we please move these helper methods to the beginning of the class declaration?

Reply:
Adding them just below the setUp func.